.github/workflows/tests.yml

```
@@ -0,0 +1,35 @@
+name: tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
```

```suggestion
        uses: actions/setup-python@v5
```

https://github.com/actions/setup-python/releases/tag/v5.1.0